package/MDAnalysis/core/selection.py

```diff
@@ -600,6 +600,38 @@ def apply(self, group):
         return group[group.aromaticities].unique
 
 
+class SmartsSelection(Selection):
+    """Select atoms based on SMARTS queries"""
+    token = 'smarts'
+
+    def __init__(self, parser, tokens):
+        pattern = []
```

Comment:
Is it simpler to enforce no white space allowed in the pattern?

Reply:
White spaces are added automatically by the parser around parentheses:
https://github.com/cbouy/mdanalysis/blob/rdkit-converter/package/MDAnalysis/core/selection.py#L1182